sopel/plugin.py

```diff
@@ -1716,16 +1716,18 @@ def require_bot_privilege(
     <.bot.Sopel.say>`, but when ``reply`` is ``True`` it uses :meth:`bot.reply()
     <.bot.Sopel.reply>` instead.
 
-    Privilege requirements are ignored in private messages.
+    Use of ``require_bot_privilege()`` implies :func:`require_chanmsg`.
 
     .. versionadded:: 7.1
+    .. versionchanged:: 8.0
+        Decorated callables no longer run in response to private messages.
     """
     def actual_decorator(function):
         @functools.wraps(function)
         def guarded(bot, trigger, *args, **kwargs):
-            # If this is a privmsg, ignore privilege requirements
+            # If this is a privmsg, do not trigger
             if trigger.is_privmsg:
-                return function(bot, trigger, *args, **kwargs)
+                return
```

Comment:
This seems like a good place for a logger message (I settled on `warning` since this is a breaking change, but I can see an argument for `debug` instead)

```suggestion
                LOGGER.warning("Skipping evaluation of channel-only callable in private message: %r", function)
                return
```

Reply:
Should be `debug`, yeah. But it makes sense to do only if the non-bot version (`require_privilege`) does the same.

Reply:
We don't have a logger for the `require_privilege`, and I don't think it is necessary: the `callable` may be already protected by the appropriate decorator, but this one is reached first because of the order of declaration.

Beside, if someone put that on the generic rule for `.*`, this will span the debug logs for no reason.

So, -0.5 here from me.